app/public.js

```diff
@@ -193,13 +197,13 @@
 
   function getImageCoverage(image) {
     let naturalArea = image.naturalSize.width * image.naturalSize.height;
-    let renderArea = image.width * image.height;
+    let renderArea = window.devicePixelRatio * window.devicePixelRatio * image.width * image.height;
```

Comment:
Because we need to apply the pixel ratio to width and height as we're calculating an area.